server/main.py

```diff
@@ -153,6 +160,111 @@ async def game_loop():
     while True:
         await asyncio.sleep(TIME_BETWEEN_ROUNDS)
         # HANDLING EACH TICK GOES HERE.
+        await game.update()
+
+        await send_updates(game.out_queue)
+
+
+async def send_updates(out_queue: asyncio.Queue):
+    """Sends all the events in the queue to their respective players."""
+    while not out_queue.empty():
+        action = await out_queue.get()
+        update: ActionUpdateMessage()
+        player_uids: int | set
+        match action:
+            case {"caster": uid}:
+                player_uids = uid
+                update = ActionUpdateMessage(
+                    type="update", message=get_action_update_message(action)
+                )
+            case {"player": uid, "direction": direction, "success": success}:
+                player_uids = uid
+                succeed = "succeed!" if success else "fail!"
+                update = ActionUpdateMessage(
+                    type="update",
+                    message=f"You try moving {direction}... and {succeed}",
+                )
+            case {"type": "room_action"}:
+                player_uids = get_room_update_uids(action)
+                update = ActionUpdateMessage(
+                    type="update", message=get_room_update_message(action)
+                )
+            case int():
+                player_uids = action
+                update = ActionUpdateMessage(
+                    type="update",
+                    message="Time passes by, but you didn't do anything this round!",
+                )
+
+        match player_uids:
+            case int():
+                # Send to a single player.
+                player_connection = connections.get(player_uids)
+                if player_connection is not None:
+                    await player_connection.send(update.json())
+                # else: player disconnected.
+            case set():
+                # Broadcast to multiple players.
+                player_connections = {
+                    connections.get(player_uid)
+                    for player_uid in player_uids
+                    if connections.get(player_uid) is not None
```

Comment:
the `is not None` here isn't necessary, just `if connections.get(player_uid)` is fine.

Reply:
True, but it doesn't hurt being a bit more explicit imo.

Reply:
Sure, but in this case (as with `if truthy` being preferred over `if truthy is True`) it's OK just to leave out `is not None`.